src/Libraries/Nop.Services/Orders/OrderProcessingService.cs

```diff
@@ -2405,7 +2423,7 @@ public virtual async Task CancelOrderAsync(Order order, bool notifyCustomer)
             //Adjust inventory
             await ReturnOrderStockAsync(order, string.Format(await _localizationService.GetResourceAsync("Admin.StockQuantityHistory.Messages.CancelOrder"), order.Id));
 
-            await _eventPublisher.PublishAsync(new OrderCancelledEvent(order));
+            await _eventPublisher.PublishAsync(new OrderStatusChangedEvent(order, prevOrderStatus));

```

Comment:
This new event raises only when the order is canceled. What about all the other status changes?